src/azul/indexer/__init__.py

```diff
@@ -121,12 +225,12 @@ def parse(cls, spec: str) -> 'SimpleSourceSpec':
         name, sep, prefix = spec.rpartition(':')
         reject(sep == '',
                'Source specifications must end in a colon followed by an optional UUID prefix')
-        return cls(prefix=prefix, name=name)
+        return cls(prefix=Prefix.parse(prefix), name=name)
 
     def __str__(self) -> str:
         """
-        >>> str(SimpleSourceSpec(prefix='12', name='foo:bar/baz'))
-        'foo:bar/baz:12'
+        >>> str(SimpleSourceSpec(prefix=Prefix.parse('12/2'), name='foo:bar/baz'))
```

Comment:
As previously stated, every class that defines the `parse`, `__str__` pair should have a doctest in `__str__` proving that one is the inverse of the other. 

To be more specific, that test should look like 

```python
>>> s = 'string_representation'
>>> s == str(Foo.parse(s))
True
```

That test alone should cover all the code in `parse` and `__str__` except for code that handles invalid inputs. If this test case leaves code uncovered in those two methods, more cases for different string repesentations may need to be added until all non-error handling code in those two methods is covered.